Improve install docs (#950)
ucfopen/UDOIT: INSTALL.md

### Option 1: Git
We strongly recommend managing the source code through Git. The benefit of this method is that you can update an existing installation of UDOIT by simply using `git pull`. It also lets you roll back to previous versions if needed. Follow these steps:

1. Install Git on your server (https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
2. Create the directory in which UDOIT will reside (e.g `UDOIT`) and navigate to it.
3. Inside the UDOIT directory, run `git clone git@github.com:ucfopen/UDOIT.git . ` (The . is important; It tells Git to download the files to the current directory.)

### Option 2: Zip File
If you prefer not to use Git, you can download the zip file of the latest release from the [Releases Page](https://github.com/ucfopen/UDOIT/releases).  Unzip it in the directory in which UDOIT will reside.

## .ENV Setup
UDOIT stores configuration variables in a `.env` file. To create it:

1. Inside the UDOIT directory, run the following command in shell:
```
cp .env.example .env
```
This command copies the `.env.example` into `.env`, creating the `.env` file in the process if it does not exist.

> 2. Open `.env` with a text editor and make the necessary changes to the following variables:

Reviewer: I assume some of the people trying to get this set up may not be text savvy, so maybe mention VS Code as an example of a text editor they can use!

	- `APP_ENV`: If you are setting up a development environment, change this to `dev` and follow the steps in [Installing Composer Dependencies](#installing-composer-dependencies) without the `--no-dev` flag to obtain all of the development packages. Otherwise, leave it as `prod`.

> 	- `DATABASE_URL`: If you are hosting UDOIT on Docker or your local machine, leave it as it is. Otherwise, change it to your database URL.

Reviewer: ```suggestion
	- `DATABASE_URL`: If you are hosting UDOIT on Docker locally, leave it as it is. Otherwise, change it to your database URL.
```
This only works if Docker is running on your local machine, not either or. Same suggestion for the following two lines, but I'm good with any other similar changes!

Author: If UDOIT is being hosted locally, doesn't that also mean the database is available too? Or are we seeing the database as a separate piece?

Reviewer: They should be separate, though I definitely misread what your comment was trying to say. Will make a small change and work on pushing this up to `dev`!

	- `APP_LMS`:
	   * `canvas` for Canvas LMS.
	   * `d2l` for D2l Brightspace LMS.
	- `JWK_BASE_URL`: If you are self-hosting Canvas, you may set it to the URL of your instance of Canvas. (Example: `JWK_BASE_URL="https://canvas.dev.myschool.edu"`)
	- `DEFAULT_LANG`: (optional) the default language of UDOIT. This is English by default.
		-  `en` for English
		-  `es` for Spanish.

## Installation

### Option 1. Docker
We provide a fast and simple way of setting up a local UDOIT instance through Docker.

1. Install [Docker Desktop](https://docs.docker.com/get-docker/). This will install Docker and Docker Compose on your system. 
	> Alternatively, you may install Docker and [Docker Compose](https://docs.docker.com/compose/install/) individually.

3. Build the Containers
If you prefer to build the containers yourself, or you are actively developing UDOIT and need to rebuild the containers to test your changes, run the following command from within the UDOIT directory:
```
    docker compose -f docker-compose.nginx.yml up
```

> 4. Wait for the containers to finish initializing. This may take over 15 minutes. If you see no output in your terminal for few minutes, proceed to the next step.

Reviewer: I'd probably add a blurb here that it stalls at step 4/5 and that's the lengthiest step. That way users know not to `cmd + C` and just wait a little longer.

Author: with (https://github.com/ucfopen/UDOIT/pull/958) changing the workflow to run the build command in daemon mode, there is now a clear indication of when the containers are initialized. There should be no need to tell users to wait in the instructions.

> 5. Once the containers are intialized, run the following command:

Reviewer: ```suggestion
5. Once the containers are initialized, run the following command:
```

Fixes typo!

```
    docker compose -f docker-compose.nginx.yml run php php bin/console doctrine:migrations:migrate
```
This applies migrations necessary to set up the database to store all UDOIT data.

Running this will give the following warning:
 > WARNING! You are about to execute a migration in database "udoit3" that could result in schema changes and data loss. Are you sure you wish to continue? (yes/no) [yes]:

> Don't worry! Simply type `yes` and proceed.

Reviewer: This feels weird after a warning that it might wreck the user's setup. I would probably omit the "Don't worry!" but the rest is great!

> You will need to run this command whenever you update to a new version of UDOIT.

> 5. UDOIT should be installed and running in your Docker cotnainer. 

Reviewer: ```suggestion
5. UDOIT should be installed and running in your Docker container. 
```

Fixes typo!

> 6. To stop UDOIT, run the following command:

Reviewer: ```suggestion
6. To stop the UDOIT containers, run the following command:
```
Taking out the section headings, I'd prefer to be more verbose in the instructions, so the user has a better idea of what they'll be executing. This'll also help us when we're troubleshooting issues for users that come to us for help.

    yarn build

## Testing successful installation

> While UDOIT only functions fully within an LMS, there is one URL publicly available you can access to test your server setup.

Reviewer: ```suggestion
While UDOIT is an LTI tool that only functions fully within an LMS, there is one URL publicly available you can access to test your server setup. Plug this into your URL bar: 
```

A little more verbosity here will help!

    <BASE_URL>/lti/config

For example, if you are setting this up on your local computer via Docker, it may look like:

    http://127.0.0.1:8000/udoit3/lti/config

## Connecting UDOIT to an LMS

> UDOIT is an LTI tool that functions fully within an LMS. 

Reviewer: ```suggestion
UDOIT is an LTI tool that functions fully within an LMS. To configure it fully within your LMS, follow the installations instructions below that apply to you.
```
Way more verbose, so I'm open to pushback on this one.

Author: I don't think either matters - I'd argue that the following step implies choosing either one, but doesn't hurt to add your suggestion. Adding it in a commit.
